### ISSUE TITLE
Error while reading a message from server. ("no core packages in the list of mods the server has enabled")

### ISSUE THREAD

**Issue by Jade-Harleyy** [Regalis11/Barotrauma#10270]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I attempted to join a friend's dedicated server through IP, but recieved an error before being prompted to enter the password.

I have double checked that no mods were enabled, and that my core package was set to "Vanilla".
File verification through Steam yielded no changes.

5+ other players were able to join the server with no issues, I was the only person to experience this error when trying to join.

Additionally, the server host attempted to restart the server multiple times to no avail, and I tried to join multiple times while the server was various states, such as in the lobby and in a level.

### Reproduction steps

1. Launch game.
2. Open server browser.
3. Attempt to join a dedicated server using the "Direct Join" feature.
4. Error is displayed, and connection fails.

### Bug prevalence

Happens every now and then

### Version

0.19.14.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
[11/06/2022 12:51:11] Error while reading a message from server. {Error in ModDownloadScreen: no core packages in the list of mods the server has enabled. Content package list received: True}
   at Barotrauma.ModDownloadScreen.Select() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Screens\ModDownloadScreen.cs:line 89
   at Barotrauma.Networking.GameClient.OnConnectionInitializationComplete() in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 1064
   at Barotrauma.Networking.LidgrenClientPeer.HandleDataMessage(NetIncomingMessage lidgrenMsg) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\Primitives\Peers\LidgrenClientPeer.cs:line 156
   at Barotrauma.Networking.LidgrenClientPeer.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\Primitives\Peers\LidgrenClientPeer.cs:line 123
   at Barotrauma.Networking.GameClient.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Networking\GameClient.cs:line 477
```

**Comment by Jade-Harleyy** [comment]
**Update:** I was able to join the server after restarting a couple times, and waiting a few minutes after before successfully joining.

**Comment by Regalis11** [comment]
It seems like this error is shown if the hashes of the core package don't match between the client and the server (e.g. if one player has an out-of-date vanilla package or has done some modifications to it). Not sure if that was what happened in this case or if there's something else wrong too, but I think we should at least fix that error message.

**Comment by juanjp600** [comment]
Should be fixed inhttps://github.com/Regalis11/Barotrauma-development/commit/17e5de471d005829a3a04deedfc758bbbd003519

**Comment by Rokvach** [comment]
Tested on the `fix/mod-error-handling-fixes` branch and found the following issue

The game seems to crash when trying to join a server which is using a core mod and you opt to download it from the host

![image](https://user-images.githubusercontent.com/7299826/203068426-d0fb60cd-e458-4367-be26-5a2645c731c1.png)

```
Barotrauma Client crash report (generated on 11/21/2022 15:30:53)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


45A76477FEB1D9B9516675F633FB1496


Game version 0.20.3.0 (ReleaseWindows, branch fix/mod-error-handling-fixes, revision aae54d4c17)
Graphics mode: 1920x1080 (Windowed)
VSync OFF
Language: English
Selected content packages: Vanilla, Reactor Health
Level seed: no level loaded
Loaded submarine: None
Selected screen: Barotrauma.ModDownloadScreen
SteamManager initialized
Client (Round hadn't been started)


System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1.7777778}
    GPU status: Normal


Exception: Could not find regular package "Nil Test Core Mod 2" (System.Exception)
Target site: Void Update(Double)
Stack trace: 
   at Barotrauma.ModDownloadScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Screens\ModDownloadScreen.cs:line 328
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 862
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
```
